File: Python Labs/com/pattern/objectpool/ClothingItemPool.py
from com.pattern.objectpool.ClothingItem import ClothingItem


class ClothingItemPool:
    def __init__(self):
        self.available_items = []
        self.in_use_items = []

        # Initialize the pool with some clothing items
        for _ in range(10):
            self.available_items.append(ClothingItem("Generic"))

    def borrow_item(self):
        if not self.available_items:
            # If no available items, create a new one or handle accordingly
            print("No available clothing items. Creating a new one.")
            return ClothingItem("Generic")
        else:
            item = self.available_items.pop(0)
            self.in_use_items.append(item)
            return item

    def return_item(self, item):
        item.return_item_to_pool()
        self.in_use_items.remove(item)
        self.available_items.append(item)

File: Java Labs/Patterns On Java/src/com/pattern/objectpool/ClothingItemPool.java
package com.pattern.objectpool;

import java.util.ArrayList;
import java.util.List;

public class ClothingItemPool {
    private List<ClothingItem> availableItems = new ArrayList<>();
    private List<ClothingItem> inUseItems = new ArrayList<>();

    public ClothingItemPool() {
        // Initialize the pool with some clothing items
        for (int i = 0; i < 10; i++) {
            availableItems.add(new ClothingItem("Generic"));
        }
    }

    public ClothingItem borrowItem() {
        if (availableItems.isEmpty()) {
            // If no available items, create a new one or handle accordingly
            System.out.println("No available clothing items. Creating a new one.");
            return new ClothingItem("Generic");
        } else {
            ClothingItem item = availableItems.remove(0);
            inUseItems.add(item);
            return item;
        }
    }

    public void returnItem(ClothingItem item) {
        item.returnItemToPool();
        inUseItems.remove(item);
        availableItems.add(item);
    }
}
